### PR TITLE
[KOGITO-8421] - [SW] Ensure that the Kogito SW builder image has the …

### PR DESCRIPTION
**Pull request by spolti**
…correct Quarkus version

- Apply review suggestions

Signed-off-by: Spolti <fspolti@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO|RHPAM-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a testcase that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- <b>(Re)run Jenkins tests</b>  
  Please add comment: <b>Jenkins [test|retest] this</b>

- <b>Prod tests</b>  
  Please add comment: <b>Jenkins (re)run [prod|Prod|PROD]</b>
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Review by ricardozanini — APPROVED** [review]

File: modules/kogito-swf-builder/added/create-app.sh [hunk 1]
```diff
@@ -28,4 +29,4 @@ export MAVEN_OPTS="$("${KOGITO_HOME}"/launch/jvm-settings.sh)"
 
 cd "${PROJECT_ARTIFACT_ID}"
 
-"${MAVEN_HOME}"/bin/mvn ${MAVEN_ARGS_APPEND} -U -B clean install -DskipTests -s "${MAVEN_SETTINGS_PATH}" -Dquarkus.container-image.build=false
+"${MAVEN_HOME}"/bin/mvn ${MAVEN_ARGS_APPEND} -Dquarkus.version="${QUARKUS_VERSION}" -U -B clean install -DskipTests -s "${MAVEN_SETTINGS_PATH}" -Dquarkus.container-image.build=false
```
Comment (radtriste) [inline modules/kogito-swf-builder/added/create-app.sh:34]: Maybe leave a comment on why the quarkus version is forced there

File: scripts/build-quarkus-app.sh [hunk 2]
```diff
@@ -52,7 +52,7 @@ mvn -U "${MAVEN_OPTIONS}" \
 
 echo "Build quarkus app"
 cd "serverless-workflow-project"
-mvn ${MAVEN_OPTIONS} -U clean install -DskipTests -Dmaven.repo.local=${mvn_local_repo} -Dquarkus.container-image.build=false
+mvn ${MAVEN_OPTIONS} -U clean install -DskipTests -Dquarkus.version="${quarkus_version}" -Dmaven.repo.local=${mvn_local_repo} -Dquarkus.container-image.build=false
```
Comment (radtriste) [inline scripts/build-quarkus-app.sh:57]: same here

**Review by radtriste — APPROVED** [review]
minor comment

File: scripts/build-quarkus-app.sh (hunk 2 above)
Comment (spolti) [inline scripts/build-quarkus-app.sh:57]:
```suggestion
# Quarkus version is enforced if some dependency pulled has older version of Quarkus set.
# This avoids to have, for example, Quarkus BOMs or orther artifacts with multiple versions.
mvn ${MAVEN_OPTIONS} -U clean install -DskipTests -Dquarkus.version="${quarkus_version}" -Dmaven.repo.local=${mvn_local_repo} -Dquarkus.container-image.build=false
```

File: modules/kogito-swf-builder/added/create-app.sh (hunk 1 above)
Comment (spolti) [inline modules/kogito-swf-builder/added/create-app.sh:34]:
```suggestion
# Quarkus version is enforced if some dependency pulled has older version of Quarkus set.
# This avoids to have, for example, Quarkus BOMs or orther artifacts with multiple versions.
"${MAVEN_HOME}"/bin/mvn ${MAVEN_ARGS_APPEND} -Dquarkus.version="${QUARKUS_VERSION}" -U -B clean install -DskipTests -s "${MAVEN_SETTINGS_PATH}" -Dquarkus.container-image.build=false
```

**Review by davidesalerno — APPROVED** [review]

**Review by radtriste — APPROVED** [review]

**Pull request merged**